tools/c7n_gcp/c7n_gcp/resources/vpc.py

```
@@ -0,0 +1,332 @@
+        pass
+
+
+@VpcServicePerimeter.action_registry.register('delete')
+class VpcServicePerimeterDelete(MethodAction):
+    """The action is used for VPC service perimeter delete.
+
+    GCP action is https://cloud.google.com/access-context-manager/docs
+                            /reference/rest/v1/accessPolicies.servicePerimeters/delete
+
+    :Example:
+
+    .. code-block:: yaml
+
+        policies:
+          - name: gcp-vpc-service-perimeter-delete
+            resource: gcp.vpc-service-perimeter
+            query:
+              - organization_id: 926683928810
+            filters:
+              - type: value
+                key: status.accessLevels
+                op: contains
+                value: accessPolicies/1016634752304/accessLevels/custodian_viewer
+            actions:
+              - type: delete
+    """
+    schema = type_schema('delete')
+    method_spec = {'op': 'delete'}
+
+    def get_resource_params(self, model, resource):
+        return {'name': resource['name']}
+
+
+@VpcServicePerimeter.action_registry.register('set')
+class VpcServicePerimeterSet(MethodAction):
+    """The action is used for VPC service perimeter patch.
+
+    GCP action is https://cloud.google.com/access-context-manager/docs
+                            /reference/rest/v1/accessPolicies.servicePerimeters/patch
+
+    Example:
+
+    .. code-block:: yaml
+
+        policies:
+          - name: gcp-vpc-service-perimeters-patch
+            resource: gcp.vpc-service-perimeter
+            query:
+              - organization_id: 926683928810
+            filters:
+              - type: value
+                key: status.accessLevels
+                op: contains
+                value: accessPolicies/1016634752304/accessLevels/custodian_viewer
+            actions:
+              - type: set
+                description: new description
+                status:
+                  resources:
+                    - projects/359546646409
```

ideally would allow projects to be set by name or folder path.

---

I'm investigating it

---

Folders cannot be used in the place. 
A list of GCP resources that are inside of the service perimeter. Currently only projects are allowed. Format: projects/{project_number} 
https://cloud.google.com/access-context-manager/docs/reference/rest/v1/accessPolicies.servicePerimeters#ServicePerimeterConfig

---

@kapilt Hello. Did that change, can you please take a look? One thing to note: if a user tries to specify a project from another organization and SA has access to that project, Custodian tries to send a request which results in an error whose details are swallowed by Custodian and the user can be confused. Checking the organization of the project prior to request requires a lot of excess API calls (folders of the organization, then subfolders of those folders, etc etc., and then retrieve only those projects which are direct descendants of org or retrieved folders), so I didn't implement that. 
We can think about extending the logging so that the error details will be directed to the output and the user will be able to figure out of what's wrong more easier.